src/Windows.Core/SIZE_T.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright © .NET Foundation and Contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace PInvoke
+{
+    using System;
+    using System.Runtime.InteropServices;
+
+    /// <summary>
+    /// The maximum number of bytes to which a pointer can point.
+    /// Use for a count that must span the full range of a pointer.
+    /// </summary>
+    /// <remarks>
+    /// This struct is intended for P/Invoke marshaling and must exactly match
+    /// the size of <see cref="IntPtr"/>
+    /// </remarks>
+    [StructLayout(LayoutKind.Sequential)]
+    public struct SIZE_T
```

Comment:
Wrapper like this does not always behave the same way as direct `IntPtr` in native calling conventions. This wrapper won't always work as the `SIZE_T` type.

Reply:
We've already added many APIs that take `SIZE_T` but just used `IntPtr` or `UIntPtr` for them so far. I think I like the idea of this struct if it makes it easier to deal with these values which are often intended to be consumable as an integer and `IntPtr` is really inconvenient for that. Is that the goal here?

Reply:
The native integers feature in C# 9 is fixing the inconvenience of using `IntPtr` as an integer.

Reply:
Awesome. Then do we need this struct at all?

Reply:
Don't think we need this struct - native ints (when they become available) should be great! @AArnott do you think you'd just generate overloads for `nint` for these kinds of methods with a new `[Friendly]` attribute ? Is there a helpful way to tag the parameter now so that it can be found easily in the future when scanning the codebase for appropriate `IntPtr`/`UIntPtr` parameters?

Reply:
That's a good idea. Do you want to add a `NativeInt` flag to the `FriendlyFlags` enum now and we'll implement the code generator for it later?
But I wonder which target framework will benefit from `nint`. Just .NET Core 5.0, perhaps? Perhaps instead we should go ahead and define `SIZE_T` if it's easier to work with than `UIntPtr`, and we can program the code generator to generate `nint` overloads wherever it sees `SIZE_T` instead. That way, every single existing application today can actually benefit from some improvement both now and in the future.

Reply:
I believe C# 9 is only [supported on .NET 5](https://devblogs.microsoft.com/dotnet/welcome-to-c-9-0/#comment-6107). I will add `FriendlyFlags.NativeInt` - yes. 

As to `SIZE_T`, you're right - that would indeed be useful for `FriendlyFlags.NativeInt` code-generation for older TFM's. 

@jkotas, you wrote: 

>Wrapper like this does not always behave the same way as direct IntPtr in native calling conventions. This wrapper won't always work as the SIZE_T type.

What would be the failure mode here, and is there a way to write a workable `SIZE_T` struct definition that would mimic `UIntPtr` in the sense that it would have the same size as native `SIZE_T`/managed `UIntPtr` and can be treated like an integer in managed code?

Reply:
> What would be the failure mode here

Hard to debug crash when `SIZE_T` is used in certain position. The specific situations where it would cause problems are platform specific. For example, believe `SIZE_T` return value cause problem on x86.

> is there a way to write a workable SIZE_T struct definition that would mimic UIntPtr 
 
There is not a way to write it with 100% fidelity.

Reply:
> C# 9 is only supported on .NET 5

As the comment said - if you know what you are doing and can avoid the unsupported features, you should be able to use native integers even on .Net Standard 2.0 and .Net Framework. It is not artificially blocked.

Reply:
> if you know what you are doing and can avoid the unsupported features, you should be able to use native integers even on .Net Standard 2.0 and .Net Framework. It is not artificially blocked.

That's how I use C# 8 on .NET Framework today. I'm fine with that. Do native ints use IntPtr under the covers, or a new type defined in .NET 5?

> Hard to debug crash when SIZE_T is used in certain position. The specific situations where it would cause problems are platform specific. For example, believe SIZE_T return value cause problem on x86.

That sounds like monsters I'd rather not invite to the party. I'm thinking maybe just using `UIntPtr` would be preferable.

Reply:
> Do native ints use IntPtr under the covers

Yup